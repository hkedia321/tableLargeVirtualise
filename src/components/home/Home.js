import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import NavHeader from 'components/navHeader';
import Card from './card';
import DataTable from 'components/datatable';
import * as utils from 'utils';
import * as requestTableDataFunctions from 'actions/actionCreators';
import Loader from 'components/loader';
import ErrorDisplay from 'components/error';

const CARD_TITLE = "Your Facebook Campaigns Data"

const fetchTableData = (props) => {
    props.requestTableDataAction.requestTableData();
}

const extractStartingDate = (tableData) => {
    if (tableData.fetched && tableData.data.length !== 0)
        return tableData.data[0].trend[0].day;
    return null;
}

const extractEndingDate = (tableData) => {
    if (tableData.fetched && tableData.data.length !== 0) {
        const trend = tableData.data[0].trend;
        return trend[trend.length-1].day;
    }
    return null;
}



const Home = (props) => {

    useEffect(() => {
        fetchTableData(props);
        // eslint-disable-next-line react-hooks/exhaustive-deps
      },[])
      const startingDate = extractStartingDate(props.tableData);
      const endingDate = extractEndingDate(props.tableData);
      const subtitle = `Showing Data From ${utils.formatDateForDisplay(startingDate)} to ${utils.formatDateForDisplay(endingDate)}`;
    return (
        <>
        <NavHeader />
        <Card 
         fetchTableData = {fetchTableData.bind(null, props)} 
         title = {CARD_TITLE}
         subtitle = {(props.tableData.fetched && props.tableData.data.length !== 0 && subtitle) || ''}
         isLoading = {!(props.tableData.fetched)}
         >
        <DataTable data = {props.tableData.data} show={props.tableData.fetched && props.tableData.data.length !== 0} />
        {props.tableData.fetched && !props.tableData.data && <ErrorDisplay message = {props.tableData.errorMsg}  />}
        {!props.tableData.fetched && <Loader /> }
        </Card>
        
        </>
    )
}

const mapStateToProps = state => ({
    tableData: state.tableData
})

const mapDispatchToProps = dispatch => {
    return { 
        requestTableDataAction: bindActionCreators(requestTableDataFunctions, dispatch)
    }
}
export default connect(mapStateToProps, mapDispatchToProps)(Home);