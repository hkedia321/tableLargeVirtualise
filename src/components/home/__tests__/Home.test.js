import React from 'react';
import { render, cleanup } from 'test-utils';
import 'jest-styled-components';
import Home, {Home as HomeWithoutConnect} from '../Home';

jest.mock('react-tooltip', () => () => <div>react-tooltip</div>);

jest.mock('highcharts-react-official', () => {
    return function DummyHighcharts(
      {options}
    ) {
      return (
        <div id="MockedHighcharts">
          {JSON.stringify(options)}
        </div>
      );
    };
  });

afterEach(cleanup);

const data = [
    {
      "groups": {
        "Campaign": {
          "metadata": {
            "name": "99733871-f7c5-4b52-9773-1ca69816fc03"
          }
        }
      },
      "trend": [
        {
          "day": "2020-06-01",
          "aj_coh_0w_ios_real_acquisition": 0,
          "aj_app_and_installs": 178,
          "aj_app_ios_installs": 0,
          "aj_coh_0w_and_real_acquisition": 14
        },
        {
          "day": "2020-06-02",
          "aj_coh_0w_ios_real_acquisition": 0,
          "aj_app_and_installs": 166,
          "aj_app_ios_installs": 0,
          "aj_coh_0w_and_real_acquisition": 21
        }
      ]
    }
  ]
const initialState ={
    tableData : {   
        data, 
        fetched: true, 
        error: null, 
        errorMsg: null 
    } 
}

 it('renders Home without crashing', () => {
    const { getByTestId } = render(<Home />);
    expect(getByTestId('home'));
})

it('renders the connected app with initialState', () => {
    const mockRequestTableDataAction = jest.fn();
    const { getByTestId } = render(<Home requestTableDataAction={mockRequestTableDataAction}/>, { initialState } );
    expect(getByTestId('home'));
})

it('matches snapshot', () => {
    const requestTableData = jest.fn();
    const {container} = render(<HomeWithoutConnect tableData={{data, fetched: true, errorMsg: null, error: null}} requestTableDataAction={{requestTableData}} />);
    expect(requestTableData).toHaveBeenCalled()
    expect(container).toMatchSnapshot();
})