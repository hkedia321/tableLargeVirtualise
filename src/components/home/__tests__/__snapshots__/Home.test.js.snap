// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
.c0 {
  background-color: #7986cb;
  padding: 15px 25px;
  border-bottom: 1px solid #d3d3d3;
  box-shadow: #d9d9d9 2px 2px 4px 0px;
  min-height: 25px;
}

.c1 {
  float: right;
  margin-top: 4px;
  cursor: pointer;
}

.c11 {
  cursor: pointer;
}

.c10 {
  padding: 10px;
  text-align: right;
  width: auto;
  color: #4b4b4b;
  font-weight: normal;
  padding-bottom: 25px;
}

.c12 {
  padding: 10px;
  text-align: right;
  width: auto;
  color: #4b4b4b;
  font-weight: normal;
  padding-bottom: 10px;
}

.c13 {
  margin: 0px;
  color: #7f7f7f;
  font-weight: normal;
}

.c8 {
  width: 100%;
  table-layout: fixed;
  height: 90vh;
  overflow-y: scroll;
  border-collapse: collapse;
}

.c9 {
  background: #f2f2f2;
  padding: 15px;
}

.c3 {
  border-bottom: solid 1px #eee;
}

.c5 {
  display: inline-block;
  margin-bottom: 5px;
}

.c6 {
  font-weight: normal;
  margin-top: 0;
  color: #767676;
  font-weight: bold;
}

.c4 {
  float: right;
  cursor: pointer;
  margin-top: 20px;
  -webkit-animation: null;
  animation: null;
}

.c2 {
  background-color: #fff;
  padding: 0 15px 25px;
  box-shadow: #eee 2px 2px;
  margin: 30px auto;
  width: 80%;
}

.c7 {
  padding: 20px 0px;
}

<div>
  <div
    data-testid="home"
  >
    <div
      class="c0"
      data-testid="navHeader"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bars fa-w-14 c1"
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="c2"
      data-testid="card"
    >
      <div
        class="c3"
        data-testid="cardHead"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-redo-alt fa-w-16 c4"
          data-icon="redo-alt"
          data-prefix="fas"
          data-tip="Reload Data"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256.455 8c66.269.119 126.437 26.233 170.859 68.685l35.715-35.715C478.149 25.851 504 36.559 504 57.941V192c0 13.255-10.745 24-24 24H345.941c-21.382 0-32.09-25.851-16.971-40.971l41.75-41.75c-30.864-28.899-70.801-44.907-113.23-45.273-92.398-.798-170.283 73.977-169.484 169.442C88.764 348.009 162.184 424 256 424c41.127 0 79.997-14.678 110.629-41.556 4.743-4.161 11.906-3.908 16.368.553l39.662 39.662c4.872 4.872 4.631 12.815-.482 17.433C378.202 479.813 319.926 504 256 504 119.034 504 8.001 392.967 8 256.002 7.999 119.193 119.646 7.755 256.455 8z"
            fill="currentColor"
          />
        </svg>
        <h3
          class="c5"
        >
          Your Facebook Campaigns' Data
        </h3>
        <h4
          class="c6"
        >
          Showing Data From 
          <span>
            1
            <sup>
              st
            </sup>
          </span>
          Jun 2020 to 
          <span>
            2
            <sup>
              nd
            </sup>
          </span>
          Jun 2020
        </h4>
        <div>
          react-tooltip
        </div>
      </div>
      <div
        class="c7"
      >
        <div
          class="dataTableWrap"
        >
          <table
            class="c8"
            data-testid="datatable"
          >
            <thead>
              <tr
                class="c9"
              >
                <th
                  class="c10"
                >
                  Campaign Id
                   
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-down fa-w-10 c11"
                    data-for="columnCollapseTooltip"
                    data-icon="caret-down"
                    data-prefix="fas"
                    data-tip="true"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                      fill="currentColor"
                    />
                  </svg>
                  <div>
                    react-tooltip
                  </div>
                </th>
                <th
                  class="c12"
                >
                  App and Installs
                   
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-down fa-w-10 c11"
                    data-for="columnCollapseTooltip"
                    data-icon="caret-down"
                    data-prefix="fas"
                    data-tip="true"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                      fill="currentColor"
                    />
                  </svg>
                  <h4
                    class="c13"
                  >
                    € 344
                  </h4>
                  <div>
                    react-tooltip
                  </div>
                </th>
                <th
                  class="c12"
                >
                  CoH Real Acquisition
                   
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-down fa-w-10 c11"
                    data-for="columnCollapseTooltip"
                    data-icon="caret-down"
                    data-prefix="fas"
                    data-tip="true"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                      fill="currentColor"
                    />
                  </svg>
                  <h4
                    class="c13"
                  >
                    € 35
                  </h4>
                  <div>
                    react-tooltip
                  </div>
                </th>
              </tr>
            </thead>
            <tbody>
              <div
                style="overflow: visible; height: 0px; width: 0px;"
              />
              <div
                class="resize-triggers"
              >
                <div
                  class="expand-trigger"
                >
                  <div
                    style="width: 1px; height: 1px;"
                  />
                </div>
                <div
                  class="contract-trigger"
                />
              </div>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
